docs/dev/app/listings.md

```diff
@@ -1,9 +1,416 @@
+
+**Fields of a Filter**
+
+- `id`: **`string`**
+
+    unique identifier of the filter's dropwdown button
+
+- `options`: **`Set<string> | Map<string, string>`**
+
+    All options for a given filter that a user can select from (derived from the listing data). This can be a map or a set. A map is used when you want the options' display-text to be different from the stored-text. If a map is used, the keys are the actual values stored, while values are the displayed value.
+
+- `placeholder`: **`string`**
+
+    The placeholder display text on the filter dropdown when no values of the current filter are selected. 
+
+- `value`: **`Set<string>`**
+
+    The selected values of the filter. This should be tracked as a state in the main component, and its corresponding setState function should be passed into `onChange` below.
+
+- `onChange`: **`(newValue: Set<string>) => void`**
+    
+    Usually of the form `newValue => setFieldFilters(newValue)`. This function is called when the values of the filter dropdown is changed, i.e. when the user selects/deselects a value of a the current filter. 
+
+- (optional) `fullText`: **`string`**
+    
+    A string that is the default display name of the filter dropdown button. Currenlty, this is only used for the "Remote/In Person" filter. 
+
+**Adding a New Filter** 
+
+Below is an example of adding a new filter called `filter1`, whose options are `string`s. 
+1. Create a new state to track the filter's selected values. 
+    - Make sure to reset the filter's state in the `resetFilters` function.
+```ts
+const [filter1Filters, setFilter1Filters] = useState(new Set<string>());
+// ... 
+const resetFilters = useCallback(() => {
+    setFilter1Filters(new Set());
+    // reset other filters ... 
+}, []);
+```
+2. Define the filter's possible options. 
+    - Many of the filters in the codebase use `useMemo` with a dependency on `listingData` since the filter options (e.g. for langauges) are dependent on the options from listings. 
+```ts
+ const filter1Options = useMemo(
+        () => {...} // get filter options from listingData 
+        [listingData],
+    );
+```
+3. Update `filteredListings` by adding an additional filter for the current filter's values. 
+``` tsx
+const filteredListings = useMemo( 
+    () => {...}, 
+    // see codebase for example of filtering based on selected filter values 
+    [listingData, filter1Filters, ...]
+)
+```
+4. Add a new `Filter` object to the array of the `filters` prop. 
+``` tsx
+ <ListingPage 
+    filters={[
+        {
+            id: 'filter1',
+            options: filter1Options,
+            value: filter1Filters,
+            onChange: newValue => setFilter1Filters(newValue),
+            placeholder: 'Filter1',
+        },
+        // other filters ... 
+    ]}
+    // other props 
+/>
+```
+
+## `ListingCard` 
+
+::: tip Description
```

Comment:
Is it okay if we remove the tip containers from the all descriptions?

Reply:
gotcha, so the description would just be plain text like this? 
<img width="570" alt="Screen Shot 2024-06-03 at 2 05 26 PM" src="https://github.com/calblueprint/immigration-justice-project/assets/64336644/4c76bb90-c55d-472e-a380-9aec0581d2bd">